src/components/organisms/UiMenu/stories/HasButton.vue

```diff
@@ -0,0 +1,175 @@
+          {{ label }}
+        </UiCheckbox>
+      </template>
+      <template #custom-option>
+        <UiIcon
+          icon="plus"
+          class="ui-button__icon has-button__custom-option-icon"
+        />
+        <UiText
+          tag="span"
+          class="ui-text--body-1-thick"
+        >
+          Didn't find chronic condition?
+        </UiText>
+        <UiText
+          tag="span"
+          class="has-button__custom-option-hint"
+        >
+          Add with your own words
+        </UiText>
+      </template>
+    </UiMenu>
+  </UiPopover>
+</template>
+
+<script setup lang="ts">
+import {
+  computed,
+  useAttrs,
+  defineOptions,
+  onMounted,
+  ref,
+  inject,
+} from 'vue';
+import {
+  UiCheckbox,
+  UiIcon,
+  UiMenu,
+  UiPopover,
+  UiText,
+} from '@infermedica/component-library';
+import { equal } from '../../../../utilities/helpers';
+
+defineOptions({ inheritAttrs: false });
+const attrs = useAttrs();
+const args = computed(() => (Object.keys(attrs).reduce((object, key) => {
+  if (key !== 'items') {
+    return {
+      ...object,
+      [key]: args[key],
+    };
+  }
+  return object;
+}, {})));
+
+const value = inject('value', []);
+const allPredefinedOptionsAreLoaded = ref(false);
+onMounted(() => {
+  setTimeout(() => {
+    allPredefinedOptionsAreLoaded.value = true;
+  }, 1000);
+});
+
+const focusedOption = ref('');
+const handleFocus = (item) => {
+  focusedOption.value = item;
+};
+
+const handleUpdateModelValue = (val) => (value['value'].includes(val)
+  ? value['value'].filter((option) => (!equal(val, option)))
+  : [
+    ...value['value'],
+    val,
+  ]);
+const handleOptionClick = (val) => {
+  value['value'] = handleUpdateModelValue(val);
+};
+const handleCustomOptionClick = () => {
+  console.log('emit cusomOtptionClick');
+};
+const itemsToRender = computed(() => {
+  const customOption = {
+    name: 'custom-option',
+    class: 'has-button__custom-option',
+    listItemAttrs: { class: 'ui-list-item ui-menu-item has-button__menu-item--has-border' },
+    onClick: () => handleCustomOptionClick(),
+    onKeyup: ({key}) => { key === 'enter' ? handleCustomOptionClick() : '' },
+  };
+  const items = (attrs.items as any).map((item) => ({
+    label: item,
+    checkboxAttrs: {
+      modelValue: value['value'],
+      value: item,
+      disabled: true,
+    },
+    onClick: () => handleOptionClick(item),
+    onFocus: () => handleFocus(item),
+  }));
+  return [
+    ...items,
+    (
+      allPredefinedOptionsAreLoaded.value
+        ? customOption
+        : {}
+    ),
+  ].map((item, index) => ({
+    ...item,
+    'aria-selected': false,
+    'aria-setsize': items.length + (
+      allPredefinedOptionsAreLoaded.value
+        ? 1
+        : 0
+    ),
+    'aria-posinet': index + 1,
+  }));
+});
+</script>
+
+<style lang="scss">
+@use "../../../../styles/mixins";
+.has-button {
+  --popover-content-padding-block: 0;
+  --popover-content-padding-inline: 0;
+
+  box-shadow: none;
```

Comment:
https://www.figma.com/design/2p8nMM7W88FU6r0kA52fAA/MGP-Component-Library?node-id=4184-91237&t=bDdIbs8g2z7SuojX-4 

please ask the design why they change the master component in this place. Master has a shadow. IMO it shouldn't be changed in this PR

```suggestion
```

Reply:
OK I’m waiting for response in case I will correct it later.